### PR TITLE
[Security] Prevent creating session in stateless firewalls

### LINKED ISSUE #51319: Session created by default handlers on stateless firewalls
### Symfony version(s) affected

6.3

### Description

Tests are broken on [that project](https://github.com/hwi/HWIOAuthBundle/pull/1938) since Symfony 6.0 and after more investigation I found that Symfony is starting the session in `DefaultAuthenticationSuccessHandler::onAuthenticationSuccess()` when `determineTargetUrl()` read the session to get the target URL, even if the firewall is marked as stateless.

This result in the error:
> Session was used while the request was declared stateless.

### How to reproduce

Setup a stateless firewall using `DefaultAuthenticationSuccessHandler`.

See failing test here: https://github.com/hwi/HWIOAuthBundle/pull/1938

### Possible Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Seb33300**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix https://github.com/symfony/symfony/issues/51319
| License       | MIT
| Doc PR        | 

Please check related issue for details.

Same as https://github.com/symfony/symfony/pull/51320 with @chalasr suggestion: https://github.com/symfony/symfony/pull/51320#discussion_r1290519241

### CODE REVIEW

**Review by chalasr — CHANGES_REQUESTED** [review]
Please add test cases with `_stateless => true` for both success and failure handlers to prevent regressions

**Comment by Seb33300** [comment]
> Please add test cases with `_stateless => true` for both success and failure handlers to prevent regressions

@chalasr tests added

**Review by stloyd — APPROVED** [review]

File: src/Symfony/Component/Security/Http/Tests/Authentication/DefaultAuthenticationFailureHandlerTest.php [hunk 1]
```diff
@@ -46,6 +46,7 @@ protected function setUp(): void
 
         $this->session = $this->createMock(SessionInterface::class);
         $this->request = $this->createMock(Request::class);
+        $this->request->attributes = new ParameterBag(['_stateless' => false]);
```
Comment (mtarld) [inline src/Symfony/Component/Security/Http/Tests/Authentication/DefaultAuthenticationFailureHandlerTest.php:49]: I don't think this is needed, as the `getBoolean` will default to `false` :slightly_smiling_face:

**Comment by stloyd** [comment]
@wouterj @chalasr Could you have a second look at this?

I can confirm the fix on HWIOAuthBundle. Before:
```
PHPUnit 9.6.11 by Sebastian Bergmann and contributors.

Testing /Users/stloyd/Documents/HWIOAuthBundle/tests/Functional

<!-- Session was used while the request was declared stateless. (500 Internal Server Error) -->
// error page printed
```

After:
```
PHPUnit 9.6.11 by Sebastian Bergmann and contributors.

Testing /Users/stloyd/Documents/HWIOAuthBundle/tests/Functional

Time: 00:00.090, Memory: 18.00 MB

OK (1 test, 4 assertions)
```

**Review by mtarld — APPROVED** [review]

File: src/Symfony/Component/Security/Http/Tests/Authentication/DefaultAuthenticationFailureHandlerTest.php (hunk 1 above)
Comment (Seb33300) [reply src/Symfony/Component/Security/Http/Tests/Authentication/DefaultAuthenticationFailureHandlerTest.php:49]: `attributes` property is not defined in the request mock so it needs to be defined at least with a `new ParameterBag()`.

And I prefer to keep the `['_stateless' => false]` explicit so we can better understand why we need `attributes` to be defined.

**Comment by stloyd** [comment]
@fabpot / @nicolas-grekas or anyone else can check this fix? I would love to have it in near 6.3 release

**Comment by Seb33300** [comment]
August is usually a period of holidays in europe so @chalasr and @wouterj may not be available this week.
Let's see next week :)

**Review by stof — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @Seb33300.

**Pull request merged**